atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/EC2HostLocationSupplier.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.pool;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import com.palantir.logsafe.Preconditions;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Returns the client's datacenter and rack derived from Cassandra's EC2Snitch.
+ * <p>
+ * AWS has an endpoint that returns the datacenter and rack (in Cassandra terms) - this request will fail if not on AWS.
+ * The reply comes in the form "datacenter"+"rack", e.g. "us-east-1a", where datacenter is "us-east-1" and rack is "a".
+ */
+public final class EC2HostLocationSupplier implements Supplier<HostLocation> {
+
+    private static final OkHttpClient client = new OkHttpClient.Builder().build();
```

Comment:
Can we use a memoized supplier here?

Reply:
Good point. This class is only ever used in conjunction with HostLocationSupplier, and that file memoizes EC2HostLocationSupplier, so it effectively is memoized, but if this class was used elsewhere then this could be a worthwhile change.

https://github.com/palantir/atlasdb/blob/617bb3e4e2452b4781eef227e109b7d616795163/atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/HostLocationSupplier.java#L40

Reply:
I don't see why not, I'm gonna try it.